cocos/ui/UIButton.cpp

```diff
@@ -821,11 +824,19 @@ void Button::setTitleFontName(const std::string& fontName)
     }
     if(FileUtils::getInstance()->isFileExist(fontName))
     {
-        TTFConfig config = _titleRenderer->getTTFConfig();
-        config.fontFilePath = fontName;
-        config.fontSize = _fontSize;
-        _titleRenderer->setTTFConfig(config);
-        _type = FontType::TTF;
+        if (fontName.find(".fnt") != std::string::npos || fontName.find(".FNT") != std::string::npos)
```

Comment:
Will not work in case of .Fnt extension. Convert fontName to lower case before search.